sql/src/main/java/io/crate/analyze/relations/RelationAnalyzer.java

```diff
@@ -214,9 +215,17 @@ protected AnalyzedRelation visitQuerySpecification(QuerySpecification node, Stat
         SelectAnalysis selectAnalysis = SelectAnalyzer.analyzeSelect(
             node.getSelect(), context.sources(), expressionAnalyzer, expressionAnalysisContext);
 
+        ExpressionAnalyzer selectAnalysisExpressionAnalyzer = new ExpressionAnalyzer(
+            functions,
+            statementContext.sessionContext(),
+            statementContext.convertParamFunction(),
+            new SelectAnalysisFieldProvider(selectAnalysis, fieldProvider),
+            new SubqueryAnalyzer(this, statementContext));
+
         List<Symbol> groupBy = analyzeGroupBy(
             selectAnalysis,
             node.getGroupBy(),
+            selectAnalysisExpressionAnalyzer,
             expressionAnalyzer,
```

Comment:
Wouldn't it be possible to replace the usage of the existing expressionAnalyzer here. The SelectAnalysisFieldProvider should have (has already?) a fallback to the regular fieldProvider - so the selectAnalysisExpressionAnalyzer in a sense is a super-set of the expressionAnalyzer and should be able to handle everything on its own